test/uri_options/srv-service-name-option.json

```diff
@@ -0,0 +1,24 @@
+{
+  "tests": [
+    {
+      "description": "SRV URI with custom srvServiceName",
+      "uri": "mongodb+srv://test22.test.build.10gen.cc/?srvServiceName=customname",
+      "valid": true,
+      "warning": false,
+      "hosts": null,
+      "auth": null,
+      "options": {
+        "srvServiceName": "customname"
```

Comment:
Does the uri_options test suite actually check the srvServiceName option here? EG would the test fail with 
`"srvServiceName": "wrongname"`

Reply:
Yes it does, it compares the value parsed from the uri with the expected value.